src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -78,6 +72,7 @@ func (r *ClusterReconciler) Reconcile(
 		resources.NewNodePortService(r.Client, &redpandaCluster, r.Scheme, log),
 		resources.NewConfigMap(r.Client, &redpandaCluster, r.Scheme, log),
 		sts,
+		resources.NewClusterResource(r.Client, &redpandaCluster, r.Scheme, svc.HeadlessServiceFQDN(), sts.Key(), r.Log),
```

Comment:
hmm I see why this pattern makes sense right now but I am not sure this is good idea in the long run. Just some thoughts:
- from general architecture perspective I don't like things we create to be mixed with the resource we own and update status, I would much better like to create another package/function/file for the status reporting and not mix it into resources to be applied
- I think at some point it won't be this easy and we'll want to update status even if something failed. Right now the status reporting is kind of .. not ideal because we write the status only if all the resources were applied correctly but will this always be the case? E.g. it's already not a case for upgrqades, but there the status is reported internally from the sts resource

Overall, I'll have to think about this. These were just some thoughts to start the discussion.

But I think my main concern is that status reporting is mixed with operator files to be applied. I don't particularly like that pattern and would much rather have separate interface/handling for it.

Reply:
Ok I will let the @dimitriscruz gives his opinion

BTW we should have an issue for:
> E.g. it's already not a case for upgrqades, but there the status is reported internally from the sts resource